crates/cargo-util/src/process_builder.rs

```diff
@@ -243,47 +243,53 @@ impl ProcessBuilder {
             .stdin(Stdio::null());
 
         let mut callback_error = None;
+        let mut stdout_pos = 0;
+        let mut stderr_pos = 0;
         let status = (|| {
             let mut child = cmd.spawn()?;
             let out = child.stdout.take().unwrap();
             let err = child.stderr.take().unwrap();
             read2(out, err, &mut |is_out, data, eof| {
+                let pos = if is_out {
+                    &mut stdout_pos
+                } else {
+                    &mut stderr_pos
+                };
                 let idx = if eof {
                     data.len()
                 } else {
-                    match data.iter().rposition(|b| *b == b'\n') {
-                        Some(i) => i + 1,
-                        None => return,
+                    match memchr::memrchr(b'\n', &data[*pos..]) {
```

Comment:
Is `memrchr` here necessary? It seems a bit odd to use it here but not use it anywhere else, e.g. below during `.lines()`. I'm sure there's a ton of other places in Cargo that does byte searching...

I would imagine that most of the speedup would come from just searching a smaller slice here?

Reply:
While there is measurable difference between `memrchr` and `rposition`, it not strictly necessary. The smaller slice is the only important part. I removed `memrchr`, for the time being.

Reply:
Indeed yeah `memchr` is much faster than `position`-based searching, but I just wanted to confirm I think it's fine to use `memchr`, but ideally we'd use it more than just one location if it mattered much.